File: hw/loza/u2/hw19Interface/JavaScript/FarmSystem.js
import { Location } from "./Location.js";
import { Cage } from "./Cage.js";
import { Chicken } from "./Chicken.js";
import { Pig } from "./Pig.js";
import { Cow } from "./Cow.js";
import { Sheep } from "./Sheep.js";

function main() {
    const xCoordinate = 10;
    const yCoordinate = 20;
    const weight = 10.4;
    const gender = "male";
    const bornOn = new Date(2025, 2, 1); // Marzo = 2
    const breed = "Holstein";

    const location = new Location(xCoordinate, yCoordinate);
    const cage = new Cage(1, "Stable for cows", 2, location);

    let farmAnimal = new Chicken(true, 0, 1, breed, bornOn, gender, true, weight, cage);
    console.log("farmAnimal --> Chicken", farmAnimal.toString());

    farmAnimal = new Pig(2, breed, bornOn, gender, false, weight, cage);
    console.log("farmAnimal --> Pig", farmAnimal.toString());

    farmAnimal = new Cow(false, weight, 3, breed, bornOn, gender, true, weight, cage);
    console.log("farmAnimal --> Cow", farmAnimal.toString());

    farmAnimal = new Sheep(bornOn, 4, breed, bornOn, gender, false, weight, cage);
    console.log("farmAnimal --> Sheep", farmAnimal.toString());
}

main();

File: hw/calle/u2/hw19Inheritance/FarmsystemJava/src/main/java/ec/edu/espe/farmsystem/view/FarmSystem.java
package ec.edu.espe.farmsystem.view;

import ec.edu.espe.farmsystem.model.Cage;
import ec.edu.espe.farmsystem.model.Chicken;
import ec.edu.espe.farmsystem.model.Cow;    
import ec.edu.espe.farmsystem.model.FarmAnimal;
import ec.edu.espe.farmsystem.model.Location;
import ec.edu.espe.farmsystem.model.Pig;    
import ec.edu.espe.farmsystem.model.Sheep;  
import java.util.Date;

public class FarmSystem{
    
    public static void main (String[]args){
    
        String breed;       
        Date bornOn;
        String gender;
        boolean isAbleToReproduce;
        float weight;
        Cage cage;
        Location location;
        int xCoordinate;
        int yCoordinate;
                    
        xCoordinate = 10;
        yCoordinate = 20;
        weight = 10.4F;
        gender = "male";
        isAbleToReproduce = false;
        
        
        bornOn = new Date(2025, 2, 1); 
        breed = "Holstein";
        
        location = new Location(xCoordinate, yCoordinate);
        cage = new Cage(1, "Corral for cows", 2, location);
        
        System.out.println("--- Farm System  ---");
        
        FarmAnimal chicken;
        chicken = new Chicken(true, 0, 1, breed, bornOn, gender, isAbleToReproduce, weight, cage);
        System.out.println("farmAnimal --> Chicken{" + chicken.toString().replace("Chicken{", "FarmAnimal --> Chicken{") );
        ((Chicken)chicken).layAnEgg();
        System.out.println("Laid Eggs: " + ((Chicken)chicken).getLaidEggs());
        System.out.println("-".repeat(30));
        
        FarmAnimal cow;
        cow = new Cow(true, 1.5F, 2, breed, bornOn, gender, isAbleToReproduce, weight, cage);
        System.out.println("farmAnimal --> Cow{" + cow.toString().replace("Cow{", "FarmAnimal --> Cow{") );
        ((Cow)cow).milk();
        System.out.println("-".repeat(30));

       
        Date lastSheeringDate = new Date(bornOn.getTime() + (10 * 24 * 60 * 60 * 1000L));
        FarmAnimal sheep;
        sheep = new Sheep(lastSheeringDate, 3, breed, bornOn, gender, isAbleToReproduce, weight, cage);
        System.out.println("farmAnimal --> Sheep{" + sheep.toString().replace("Sheep{", "FarmAnimal --> Sheep{") );
        ((Sheep)sheep).cutWhool();
        System.out.println("-".repeat(30));

        FarmAnimal pig;
        pig = new Pig(4, breed, bornOn, gender, isAbleToReproduce, weight, cage);
        System.out.println("farmAnimal --> Pig{" + pig.toString().replace("Pig{", "FarmAnimal --> Pig{") );
    }
}